test/external_tests.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2023 solidity contributors.
+# ------------------------------------------------------------------------------
+
+from argparse import ArgumentParser, Namespace
+import os
+from pathlib import Path
+import sys
+import subprocess
+
+EXTERNAL_TESTS_DIR = Path(__file__).parent / "externalTests"
+
+
+class ExternalTestNotFound(Exception):
+    pass
+
+
+def detect_external_tests() -> dict:
+    return {
+        file_path.stem: file_path
+        for file_path in Path(EXTERNAL_TESTS_DIR).iterdir()
+        if file_path.is_file() and file_path.suffix == ".sh"
+    }
+
+
+def display_available_external_tests(_) -> int:
+    print("Available external tests:")
+    print(*detect_external_tests().keys())
+    return os.EX_OK
+
+
+def run_test_scripts(solc_binary_type: str, solc_binary_path: Path, tests: dict) -> int:
+    for test_name, test_script_path in tests.items():
+        print(f"Running {test_name} external test...")
+        ret = subprocess.run(
+            f"{test_script_path} {solc_binary_type} {solc_binary_path}",
+            shell=True,
+            executable="/bin/bash",
+            check=True,
+        )
```

Comment:
We don't need to use the shell here:
```suggestion
        return subprocess.run(
            [test_script_path, solc_binary_type, solc_binary_path],            
            check=True
        ).returncode
```